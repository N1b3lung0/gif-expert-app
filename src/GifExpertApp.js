import React from 'react'

export const GifExpertApp = () => {
    return (
        <>
            <h2>GefExpertApp</h2>
            <hr/>
        </>
    )
}
