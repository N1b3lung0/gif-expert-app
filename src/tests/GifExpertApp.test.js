import React from 'react'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Testing GifExpertApp component', () => {
    
    test('should show the GifExpertApp component correctly', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
        const categories = ['Friends', 'Game of Thrones'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
