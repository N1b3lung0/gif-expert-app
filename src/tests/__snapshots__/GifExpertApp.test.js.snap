// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Friends"
      key="Friends"
    />
    <GifGrid
      category="Game of Thrones"
      key="Game of Thrones"
    />
  </ol>
</Fragment>
`;

exports[`Testing GifExpertApp component should show the GifExpertApp component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
